packages/iTwinUI-react/src/core/NotificationMarker/NotificationMarker.tsx

```diff
@@ -0,0 +1,78 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { CommonProps, useTheme } from '../utils';
+import cx from 'classnames';
+import '@itwin/itwinui-css/css/notification-marker.css';
+
+export type NotificationMarkerProps = {
+  /**
+   * Content of the NotificationMarker.
+   */
+  children: React.ReactNode;
+  /**
+   * Type of notification
+   *
+   * 'primary' = blue,
+   * 'positive' = green,
+   * 'warning' = orange,
+   * 'negative' = red,
+   * @default 'primary'
+   */
+  type?: 'primary' | 'positive' | 'warning' | 'negative';
+  /**
+   * Set this to true for important notifications
+   * @default false
+   */
+  urgent?: boolean;
+  /**
+   * Set this programmatically to false when you just want to render the passed children without the notification
+   * @default true
+   * @example
+   * let [newMessagesCount, ...] = useState(0);
+   * ...
+   * <NotificationMarker active={newMessagesCount > 0}>
+   *   <SvgNotification />
+   * </NotificationMarker>
+   */
+  active?: boolean;
+} & Omit<CommonProps, 'title'>;
+
+/**
+ * A small notification circle to the top-right of the passed children prop. This can be applied to pretty much anything but mostly intended for icons within default / borderless buttons.
+ * @example
+ * // Primary Intended Use-case
+ * <IconButton styleType='borderless'>
+ *   <NotificationMarker><SvgNotification /></NotificationMarker>
+ * </IconButton>
+ * @example
+ * <NotificationMarker type='positive' urgent={true}>Live</NotificationMarker>
+ */
+export const NotificationMarker = (props: NotificationMarkerProps) => {
+  const {
+    className,
+    children,
+    type = 'primary',
+    urgent = false,
+    active = true,
+    ...rest
+  } = props;
+  useTheme();
+  return (
+    <div
```

Comment:
divs inside buttons is invalid markup. Does it still work with `<span>`?

Reply:
To confirm, are you referring to what is mentioned in this [StackOverflow answer](https://stackoverflow.com/a/12982293/11547064)? This concept is new to me. I have a few questions:

* What is "markup" exactly?
* What is the meaning of "valid" markup?
* Why `<div>` inside `<button>` is invalid markup?
* How to know what is valid markup and what isn't? Do we always have to put it in a [validator tool](https://validator.w3.org/#validate_by_input+with_options) like the one mentioned in the answer? Or is there a website/source with rules for valid markup?

Reply:
markup is just html. "valid" means it's not allowed and will not be properly announced by assistive technologies (e.g. screen readers). and yes you need to validate it through a tool if you're not sure.
https://validator.w3.org/docs/help.html#validation_basics

there is an html spec but you're not expected to read the whole thing all at once 😄 https://html.spec.whatwg.org/

Reply:
**[DONE]** Changed div to span and it seems to do the same effect.